### PR TITLE
Tower system implemented

### PR DESCRIPTION
**Pull request by milosmilosevic13**
Added a tower system that takes care of building towers by checking if they are done, and attacking enemies that are close. Also added the Chanting mage state that is responsible for increasing the towers' building progress.

### CODE REVIEW

File: source/dagger/gameplay/team_game/towers.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            sprite.pivot = { 0,0.25f };
+            sprite.scale = { 2,2 };
+        }
+
+        if (range_.targetFound) tower_.currentAction = EAction::Attacking;
+
+        AnimatorPlay(animator_, "ancient_defenders:towers:" + tower_.type);
+        if (tower_.currentAction == EAction::Idling) {}
+        else if (tower_.currentAction == EAction::Attacking) {
+            Engine::Registry().get<Health>(range_.target).currentHealth -= 15.0f*Engine::DeltaTime();
```
Comment (DPrivalov) [inline source/dagger/gameplay/team_game/towers.cpp:46]: 15.0f should be some const or variable

File: source/dagger/gameplay/team_game/towers.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        static Float32 constructionGoal;
+
+        EAction currentAction = EAction::Idling;
+        String type;
+    };
+
+
+    struct Tower {
+        Entity entity;
+        Transform & coordinates;
```
Comment (DPrivalov) [inline source/dagger/gameplay/team_game/towers.h:32]: don't save components, get them from entity every frame.

**Review by DPrivalov — CHANGES_REQUESTED** [review]

File: source/dagger/gameplay/team_game/towers.cpp (hunk 1 above)
Comment (milosmilosevic13) [reply source/dagger/gameplay/team_game/towers.cpp:46]: This part of the code will be re written, this is just a placeholder for testing things.

**Comment by milosmilosevic13** [comment]
A lot of this code was changed when controls got added, due to a number of bugs that have appeared; this will be added along with the controls and a bunch of fixes together in a new pull request.

**Pull request closed without merging**